cloud/amazon/sns_topic.py

```diff
@@ -95,11 +94,15 @@ def canonicalize_endpoint(protocol, endpoint):
 
 
 
-def get_all_topics(connection):
+def get_all_topics(connection, module):
     next_token = None
     topics = []
     while True:
-        response = connection.get_all_topics(next_token)
+        try:
+            response = connection.get_all_topics(next_token)
+        except BotoServerError as e:
```

Comment:
This syntax is unfortunately not Python 2.4 compatible, and while Ansible doesn't check the cloud directory for Python 2.4 compatibility, I'm still being asked to keep my contributions to cloud/ compatible with Python 2.4 (see #1429). `sys.exc_info()[1]` is, I believe, the best way to do it which has the broadest Python compatibility (2.4 through 3.x).